cmd/fpga_crihook/main.go

```diff
@@ -44,54 +48,56 @@ func decodeJSONStream(reader io.Reader) (map[string]interface{}, error) {
 }
 
 type hookEnv struct {
-	bitStreamDir string
-	config       string
-	execer       utilsexec.Interface
+	bitStreamDir  string
+	config        string
+	execer        utilsexec.Interface
+	aFUIDTemplate string
 }
 
-func newHookEnv(bitStreamDir string, config string, execer utilsexec.Interface) *hookEnv {
+func newHookEnv(bitStreamDir string, config string, execer utilsexec.Interface, aFUIDTemplate string) *hookEnv {
 	return &hookEnv{
 		bitStreamDir,
 		config,
 		execer,
+		aFUIDTemplate,
 	}
 }
 
 func canonize(uuid string) string {
 	return strings.ToLower(strings.Replace(uuid, "-", "", -1))
 }
 
-func (he *hookEnv) getFPGAParams(reader io.Reader) (string, string, error) {
+func (he *hookEnv) getFPGAParams(reader io.Reader) (string, string, string, error) {
```

Comment:
I guess now these 3 strings in the function's return would be better joined into a struct like
```
type fpgaParams {
    region string
    afu string
    devNum string
}
```

Reply:
this caused 208 lines of changes so far. If I'd know that I'd not even start. It doesn't worth it in my opinion.